docs/entity_creation.md

```diff
@@ -0,0 +1,203 @@
+# Entity Creation
+
+
+
+Expanding the current selection of indexed entities is a way of increasing the coverage of queryable network interactions. 
+
+### Implementation
+
+
+#### Adding the entity schema
+
+in `schema.graphql`:
+
+The graphql schema is amended to include the new entity, with any fields that are required by the future scope of querying. These attributes are preferably only those required inn order to conserve storage.
+```
+0  type TestEntity @entity {
+1  		id: ID!
+2   	entity_field_a: String! @index
+3   	entity_field_b: String! @index
+4   	entity_field_c: String!
+5   	message: Message!
+6   	transaction: Transaction!
+7   	block: Block!
+8  }
+```
+
+- L0 - Entity is initialised with the `@entity` tag
+
+- L1 - Entity is afforded a unique identifier, ID
+
+- L2-4 - Entity attributes and their types are introduced, some with the `@index` tag to allow more complex querying
+
+- L5-8 - Relevant message, transaction and block are provided to the entity
+
+Updating the system to include these changes can be done by running the following: 
+```
+yarn codegen
+```
+
+
+#### Implementing the entity handler
+
+in `/src/mappings/your_entity_category/entity_name.ts`
+
+This handler function will be configured to trigger whenever the indexer detects the desired interaction with the entity:
+
+```
+0	export async function handleYourEntityName(event: CosmosEvent): Promise<void> {
+1		logger.info(`confirmation that your handler is triggered`);
+2
+3		const id = messageId(event.msg);
+4  		const decoded_msg = event.msg.msg.decodedMsg;
+5  		const entity_field_a = decoded_msg.entity_field_a;
```

Comment:
Let's add a code comment about explicitly avoiding use of destructuring for variable assignment and prefer that other thing that I can't remember the name of right now.